src/dotnet/commands/dotnet-build3/MSBuildForwardingApp.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Microsoft.DotNet.Cli.Utils;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Cli
+{
+    public class MSBuildForwardingApp : ForwardingApp
+    {
+        private static readonly string s_msbuildExeName = "msbuild.exe";
+
+        public MSBuildForwardingApp(string[] argsToForward) :
+            base(GetMSBuildExePath(), 
+                argsToForward,
+                depsFile: GetDepsFile(),
+                runtimeConfig: GetRuntimeConfig(),
+                environment: GetEnvironment()) { }
+
+        private static Dictionary<string, string> GetEnvironment()
+        {
+            return new Dictionary<string, string>
+            {
+                { "MSBuildExtensionsPath", GetDotnetMSBuildTargetsPath() },
+                { "DotnetHostPath", GetHostPath() }
```

Comment:
@cdmihai @rainersigwald Calling this out because the cli needs to supply the path to the currently used host (which is not necessarily universal since multiple clis can be installed in different locations). 

Although this works, I'm not sure what the **right** thing to do here is. How will msbuild find this when it's not executed via the cli?